src/main.cpp

```diff
@@ -658,14 +658,8 @@ bool IsStandardTx(const CTransaction& tx, string& reason)
 
 bool IsFinalTx(const CTransaction &tx, int nBlockHeight, int64_t nBlockTime)
 {
-    AssertLockHeld(cs_main);
-    // Time based nLockTime implemented in 0.1.6
     if (tx.nLockTime == 0)
         return true;
-    if (nBlockHeight == 0)
```

Comment:
> Then why maintain a global-state-dependent function at all?

Huh? The global state-dependent function is for the convenience of callers, who just need to know if the tx will be final by the time the next block comes.

> If we're going to review all the calls and change all those lines, let's just have the stateless version of it.

Having only the stateless IsFinalTx() as an option results in most of those calls having useless - and probably soon to be obsolete - Tip()->nHeight, GetAdjustedTime() boilerplate.

> And at that point, you're only changing one function, shouldn't the criterion "if you change functionality, change the name of the function" precisely apply here?

The functionality of mixing up a consensus-critical and non-consensus-critical function in one is simply broken.

> No, before the null values were replaced with values from the global state, now they don't.
They clearly function differently (even if you're taking care of it not mattering by adapting things on the caller, the function itself is NOT functionally equivalent to the previous stateful one).

Again, the previous design of IsFinalTx() is simply broken and shouldn't have happened; from the point of view of the important stuff - the consensus-critical code - the functionality hasn't changed changed. Fortunately we never had a situation where that *broken design* bit us in the ass so in spite of it we didn't end up with a forking bug; if the functionality had changed from the point of view of consensus-critical code it'd be a guaranteed forking bug because it's comparing nLockTime to state that is *not the same on every node*.